src/Raven.Server/Documents/Indexes/SharedIndexJournals.cs

```diff
@@ -50,6 +50,11 @@ public SharedIndexJournals(DocumentDatabase documentDatabase)
         options.MinimumSharedJournalsMergeCount = documentDatabase.Configuration.Storage.MinimumSharedJournalsMergeCount;
         options.MaxLogFileSize = documentDatabase.Configuration.Storage.MaxJournalFileSize.GetValue(SizeUnit.Bytes);
 
+        options.OnNonDurableFileSystemError += documentDatabase.HandleNonDurableFileSystemError;
+        options.OnRecoveryError += (s, e) => documentDatabase.HandleOnIndexRecoveryError("JournalsForIndexing", s, e);
+        options.OnIntegrityErrorOfAlreadySyncedData += (s, e) => documentDatabase.HandleOnIndexIntegrityErrorOfAlreadySyncedData("JournalsForIndexing", s, e);
+        options.OnRecoverableFailure += documentDatabase.HandleRecoverableFailure;
```

Comment:
I'm surprised that it was missing. Can we have some dedicated tests which verify that those are properly hooked up?

Reply:
done